### PR TITLE
[FIX] sale_pdf_quote_builder: correctly fill editable fields

### PR DESCRIPTION
**Pull request by Demesmaeker**
PDF written with scribus (and some other software) have a problem when filling fields: the output does not work as expected. The replaced value is there, but hidden behind a blue overlay, shown only when clicking on it.

We now show the field value and ensure filled fields are read only.

Additionally, some readers only allow a single value per field name. Even if the values were different on the documents, only one would be shown. We now rename the fields to ensure they are different when they have different values.

task-3626047

### CODE REVIEW

**Comment by llabusch93** [comment]
Hello,

Do you have any updates on when this merge will be completed? Since the previous quote builder, which allowed for editing the portal template, has been removed and the PDF quote builder isn't functioning, we currently lack any means to style quotes.

Given that this was prominently featured as a new and exciting feature in the keynote, I believe it should be addressed with high priority, don't you agree?

**Comment by Demesmaeker** [comment]
Hello @llabusch93 
We indeed take this in high regards. This task is under testing at the moment, and should be available soon =]
Have a nice day!

File: odoo/tools/pdf.py [hunk 1]
```diff
@@ -93,34 +93,48 @@ def merge_pdf(pdf_data):
...
-    for page in range(0, reader.getNumPages()):
-        page = reader.getPage(page)
-        writer.addPage(page)
-        try:
+    # This solves a known problem with PyPDF2, where with some pdf software, forms fields aren't
+    # correctly filled until the user click on it, see: https://github.com/py-pdf/pypdf/issues/355
+    try:
+        writer.set_need_appearances_writer()
+        is_upper_version_pypdf2 = True
+    except AttributeError:  # This method was renamed in PyPDF2 2.0
```
Comment (Feyensv) [inline odoo/tools/pdf.py:108]: `hasattr` instead of try/except ?

File: addons/sale_pdf_quote_builder/models/ir_actions_report.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
-# -*- coding: utf-8 -*-
 import io
```
Comment (Feyensv) [inline addons/sale_pdf_quote_builder/models/ir_actions_report.py:1]: license

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

(+ as explained, this requires test coverage soon)

**Comment by Demesmaeker** [comment]
@robodoo r+

**Comment by Demesmaeker** [comment]
@robodoo r+

**Pull request closed without merging**